services/dy-modeling/source/class/qxapp/components/AvailableServices.js

```diff
@@ -0,0 +1,216 @@
+      this.__dodecaBtn.addListener("execute", this.__onAddDodecaRequested.bind(this));
+
+      this.__splineBtn = new qx.ui.toolbar.RadioButton(this.tr("Add Spline"));
+      this.__splineBtn.addListener("execute", this.__onAddSplineRequested.bind(this));
+
+      drawingPart.add(this.__sphereBtn);
+      drawingPart.add(this.__blockBtn);
+      drawingPart.add(this.__cylinderBtn);
+      drawingPart.add(this.__dodecaBtn);
+      drawingPart.add(this.__splineBtn);
+
+      let drawingGroup = new qx.ui.form.RadioGroup(this.__sphereBtn, this.__blockBtn, this.__cylinderBtn, this.__dodecaBtn, this.__splineBtn);
+      drawingGroup.setAllowEmptySelection(true);
+      drawingGroup.setSelection([]);
+
+      // Boolean operations
+      let menuPart = new qx.ui.toolbar.Part();
+      toolbar.add(menuPart);
+
+      let booleanMenu = new qx.ui.toolbar.MenuButton("Boolean operations");
+      booleanMenu.setMenu(this.__getBooleanMenu());
+      menuPart.add(booleanMenu);
+
+      return frame;
+    },
+
+    __getBooleanMenu: function() {
+      let menu = new qx.ui.menu.Menu();
+
+      let uniteButton = new qx.ui.menu.Button(this.tr("Unite"));
+      let intersectButton = new qx.ui.menu.Button(this.tr("Intersect"));
+      let substractButton = new qx.ui.menu.Button(this.tr("Substract"));
+
+      uniteButton.addListener("execute", this.__onBooleanUniteRequested.bind(this));
+      intersectButton.addListener("execute", this.__onBooleanIntersectRequested.bind(this));
+      substractButton.addListener("execute", this.__onBooleanSubstractRequested.bind(this));
+
+      menu.add(uniteButton);
+      menu.add(intersectButton);
+      menu.add(substractButton);
+
+      return menu;
+    },
+
+    __onAddSphereRequested: function() {
+      this.fireDataEvent("newSphereRequested", this.__sphereBtn.getValue());
+    },
+
+    __onAddBlockRequested: function() {
+      this.fireDataEvent("newBlockRequested", this.__blockBtn.getValue());
+    },
+
+    _onAddCylinderRequested: function() {
+      this.fireDataEvent("newCylinderRequested", this.__cylinderBtn.getValue());
+    },
+
+    __onAddDodecaRequested: function() {
+      this.fireDataEvent("newDodecaRequested", this.__dodecaBtn.getValue());
+    },
+
+    __onAddSplineRequested: function() {
+      this.fireDataEvent("newSplineRequested", this.__splineBtn.getValue());
+    },
+
+    __onMoveToolRequested: function() {
+      this.fireDataEvent("moveToolRequested", this.__moveBtn.getValue());
+    },
+
+    __onRotateToolRequested: function() {
+      this.fireDataEvent("rotateToolRequested", this.__rotateBtn.getValue());
+    },
+
+    __onBooleanUniteRequested: function() {
+      this.fireDataEvent("booleanOperationRequested", 0);
+    },
+
+    __onBooleanIntersectRequested: function() {
+      this.fireDataEvent("booleanOperationRequested", 1);
+    },
+
+    __onBooleanSubstractRequested: function() {
+      this.fireDataEvent("booleanOperationRequested", 2);
+    }
+  }
```

Comment:
such highly repetitive code makes me itchy

Reply:
:-D can you provide some guidance today?

Reply:
I'll try :)

Reply:
Right, the client code is far from being optimum (it is the outcome of the prototype building period). The idea is to have it in simcore so that it can be used in the platform building process.